src/common/random.h

```diff
@@ -93,15 +95,24 @@ class ColumnSampler {
     if (colsample == 1.0f) return p_features;
     const auto& features = *p_features;
     CHECK_GT(features.size(), 0);
-    int n = std::max(1, static_cast<int>(colsample * features.size()));
+    unsigned long n = std::max(1, static_cast<int>(colsample * features.size()));
     auto p_new_features = std::make_shared<std::vector<int>>();
     auto& new_features = *p_new_features;
     new_features.resize(features.size());
     std::copy(features.begin(), features.end(), new_features.begin());
     std::shuffle(new_features.begin(), new_features.end(), common::GlobalRandom());
-    new_features.resize(n);
-    std::sort(new_features.begin(), new_features.end());
-
+    new_features.resize(static_cast<unsigned long>(n));
+    // std::sort(new_features.begin(), new_features.end());
```

Comment:
A bit confused by the old code here, was there any reason to sort just after we had shuffled the features?

Then could you explain why we need the ser/deser compared to what was happening before?